### PR TITLE
Update for SpongeAPI changes

### PR DESCRIPTION
**Pull request by DDoS**
- Implements some of the new changes
- Fixes a copy-paste error in MixinChunk

See: https://github.com/SpongePowered/SpongeAPI/pull/484

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/mixin/core/world/MixinWorld.java [hunk 1]
```diff
@@ -182,12 +184,39 @@ public String getName() {
...
+
+    @Override
+    public void setBlock(int x, int y, int z, BlockState block) {
+        net.minecraft.world.World world = ((net.minecraft.world.World) (Object) this);
+        // 0 is no notify flag. For now not going to notify nearby blocks of update.
+        if (block instanceof IBlockState) {
+            world.setBlockState(new BlockPos(x, y, z), (IBlockState) block, 0);
+        } else {
+            // TODO: Need to figure out what is sensible for other BlockState implementing classes.
+            throw new UnsupportedOperationException();
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/mod/mixin/core/world/MixinWorld.java]: There should be some kind of error message here.

Comment (DDoS) [inline src/main/java/org/spongepowered/mod/mixin/core/world/MixinWorld.java]: That was a copy paste from BlockWrapper#getState(). I can add it to both maybe.

**Comment by me4502** [comment]
May also be worth hooking the new NeighborNotifyEvent from Forge to BlockUpdateEvent

**Comment by DDoS** [comment]
Could you elaborate on this please? The goal of the PR is just to update for my changes in the API, it's not really for new implementation.

**Comment by me4502** [comment]
That was just modified in the api to be easier to implement

**Comment by DDoS** [comment]
Don't merge yet, I can reduce some code duplication.

**Comment by DDoS** [comment]
Fixed!

**Comment by me4502** [comment]
Merged.

**Pull request closed without merging**